instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/OpenTelemetryMeterRegistry.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.micrometer.v1_5;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.FunctionTimer;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.core.instrument.Measurement;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.HistogramGauges;
+import io.micrometer.core.instrument.distribution.pause.PauseDetector;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import java.util.concurrent.TimeUnit;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToLongFunction;
+
+public final class OpenTelemetryMeterRegistry extends MeterRegistry {
+
+  private static final String INSTRUMENTATION_NAME = "io.opentelemetry.micrometer-1.5";
+
+  public static MeterRegistry create() {
+    OpenTelemetryMeterRegistry openTelemetryMeterRegistry =
+        new OpenTelemetryMeterRegistry(
+            Clock.SYSTEM, GlobalOpenTelemetry.get().getMeterProvider().get(INSTRUMENTATION_NAME));
+    openTelemetryMeterRegistry.config().onMeterRemoved(OpenTelemetryMeterRegistry::onMeterRemoved);
+    return openTelemetryMeterRegistry;
+  }
+
+  private final io.opentelemetry.api.metrics.Meter otelMeter;
+  private final AsyncInstrumentRegistry asyncInstrumentRegistry;
+
+  private OpenTelemetryMeterRegistry(Clock clock, io.opentelemetry.api.metrics.Meter otelMeter) {
+    super(clock);
+    this.otelMeter = otelMeter;
+    this.asyncInstrumentRegistry = new AsyncInstrumentRegistry(otelMeter);
+  }
+
+  @Override
+  protected <T> Gauge newGauge(Meter.Id id, T t, ToDoubleFunction<T> toDoubleFunction) {
```

Comment:
Could use `@Nullable` annotations and nullaway throughout. For example, `t` is nullable here, and propagates in a lot of places which make it easy to forget.